### PR TITLE
Migrate all v2 resources to v3.

### LINKED ISSUE #97: Remove support for envoy < 1.14.0-dev
Currently we use envoy 1.14.0-dev (commit 6c2137468c25d167dbbe4719b0ecaf343bfb4233), but we keep backward compatibility with envoy 1.13.0-dev (commit b7bef67c256090919a4585a1a06c42f15d640a09) (excluding incoming permissions, which don't work on that version).

When the backward compatibility will be not needed, remove the deprecated code and tests checking the compatibility (all marked with TODO comments).

### PR DESCRIPTION
**Pull request by slonka**
Fixes #97

### CODE REVIEW

**Review by lukidzi — APPROVED** [review]

File: envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java [hunk 1]
```diff
@@ -519,8 +521,13 @@ protected static void assertThatWatchReceivesSnapshot(WatchAndTracker watchAndTr
 
         assertThat(response).isNotNull();
         assertThat(response.version()).isEqualTo(snapshot.version(watchAndTracker.watch.request().getTypeUrl()));
-        assertThat(response.resources().toArray(new Message[0]))
-                .containsExactlyElementsOf(snapshot.resources(watchAndTracker.watch.request().getTypeUrl()).values());
+        Message[] responseValues = response.resources().toArray(new Message[0]);
```
Comment (slonka) [inline envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java:524]: I do not know why but the `containsExactlyElementsOf` stopped working :/ even though the values underneath are the same. I converted this to an equivalent assertion but please pay attention here.

File: envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/RoutesAssertions.kt [hunk 2]
```diff
@@ -45,13 +45,13 @@ fun RouteAction.hasCustomRequestTimeout(requestTimeout: Duration): RouteAction {
     return this
 }
 
-fun RouteAction.autoHostRewriteHeaderIsEmpty(): RouteAction {
-    assertThat(this.autoHostRewriteHeader).isEmpty()
+fun RouteAction.hostRewriteHeaderIsEmpty(): RouteAction {
```
Comment (slonka) [inline envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/RoutesAssertions.kt:48]: this was renamed with backwards compatiblity: https://github.com/envoyproxy/envoy/blob/d321ede4d262cc2259064337be7b407e463ede24/generated_api_shadow/envoy/api/v2/route/route_components.proto#L869-L872

File: envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt [hunk 3]
```diff
@@ -95,7 +95,7 @@ class SnapshotDebugController(controlPlane: ControlPlane) {
             .add(Struct.getDescriptor())
             .add(Value.getDescriptor())
             .add(RBAC.getDescriptor())
-            .add(FilterRBAC.getDescriptor())
+            .add(RBACFilter.getDescriptor())
```
Comment (slonka) [inline envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt:98]: Renamed for consistency in RBAC filter factory.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ssl/EnvoyHttpsDependencyTest.kt [hunk 4]
```diff
@@ -21,22 +21,6 @@ class EnvoyCurrentVersionHttpsDependencyTest : EnvoyHttpsDependencyTest() {
     }
 }
 
-// TODO(https://github.com/allegro/envoy-control/issues/97) - remove when envoy < 1.14.0-dev will be not supported
```
Comment (slonka) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ssl/EnvoyHttpsDependencyTest.kt:24]: We no longer have any Envoy-s with this version on production.

File: envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java (hunk 1 above)
Comment (MarcinFalkowski) [reply envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java:524]: I think debugging using breakpoint, line by linem will tell us where where the problem is.

Comment (MarcinFalkowski) [reply envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java:524]: Maybe the lack of `toArray` for snapshot.resources is the problem?

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ssl/EnvoyHttpsDependencyTest.kt (hunk 4 above)
Comment (MarcinFalkowski) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ssl/EnvoyHttpsDependencyTest.kt:24]: There is also a code that needs to be removed in EnvoyClustersFactory.kt

Comment (MarcinFalkowski) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/ssl/EnvoyHttpsDependencyTest.kt:24]: You can also add "Fixes #97" to the description

**Review by MarcinFalkowski — DISMISSED** [review]

**Comment by pzmi** [comment]
Looks like this PR got quite outdated. How about we close this one and get back to it with a fresh one when we're ready?

**Pull request closed without merging**